### DISCUSSION TITLE
Improve Processors Groups?

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1302] [Dev]
Currently, we have Groups for the processors listed here https://github.com/apache/streampipes/blob/dev/streampipes-model/src/main/java/org/apache/streampipes/model/DataProcessorType.java

Is it a good idea to set a processor only to a single category or would it be better to add a processor to a group list?

For example, a processor with an aggregation implementation (that means some form of window function) could be more specific in different groups, or also be some form of filter.
There are multiple other examples, I think.

I think the main categories could also be divided into 
stream data integration
stream analytics

Altogether, the user could search and find/use processors for the requirements faster.

What do you think?

**Comment by bossenti** [comment]
I guess technically it would be rather easy to allow multiple domain assignments.
What do we want to gain from it? Do we want to two foster this information in the UI?
Without introducing changes in the UI this doesn't have a real impact, does it?
